### PR TITLE
Introduce data stiching

### PR DESCRIPTION
**Pull request by drillprop**
Fix https://github.com/Joystream/atlas/issues/1540

just FYI you need to add query node URL to env 
```
QUERY_NODE_URL=https://sumer-dev-2.joystream.app/query/server/graphql
```

### CODE REVIEW

File: tests/featuredContent.test.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ describe('Featured content resolver', () => {
   beforeEach(async () => {
     mongoose = await connectMongoose(process.env.MONGO_URL!)
     aggregates = await buildAggregates()
-    server = await createServer(mongoose, aggregates)
+    server = await createServer(mongoose, aggregates, 'https://sumer-dev-2.joystream.app/query/server/graphql')
```
Comment (drillprop) [inline tests/featuredContent.test.ts:38]: I'm guessing that I'm not supposed to pass this URL to tests, but I didn't know how to do it properly. Feel free to add any suggestion

File: package.json [hunk 2]
```diff
@@ -30,11 +30,18 @@
...
+    "@graphql-tools/schema": "^8.3.1",
+    "@graphql-tools/stitch": "^8.4.1",
+    "@graphql-tools/url-loader": "^7.5.2",
+    "@graphql-tools/wrap": "^8.3.1",
     "@typegoose/auto-increment": "^1.0.0",
     "@typegoose/typegoose": "^9.1.0",
     "apollo-server-core": "^3.4.0",
     "apollo-server-express": "^3.4.0",
     "class-validator": "^0.13.1",
+    "cross-undici-fetch": "^0.0.20",
```
Comment (kdembler) [inline package.json:44]: What is this package? Seems it's in a very early version, couldn't we use something more popular?

File: tests/featuredContent.test.ts (hunk 1 above)
Comment (kdembler) [reply tests/featuredContent.test.ts:38]: Hmm maybe we require this to come from the env as well? Not sure yet

File: package.json (hunk 2 above)
Comment (drillprop) [reply package.json]: Actually is no longer needed, removed

**Comment by kdembler** [comment]
Okay, I'm starting to get more deeply into it. Here are some of the thoughts:

1. Really great job on this, this surely required a lot of work to get right

2. We need to clean up `queryNodeStitchingResolvers.ts`, it's 500 lines long at this point and it's really hard to find what you're looking for. I suggest so make this into a directory and have separate files for:
- video resolvers
- channel resolvers
- featured content resolvers
- helper function

I think doing this will also help us better notice places for potential improvements

3. I've noticed that in some featured content resolvers we use `context` and aggregates directly. Maybe we could follow the same approach for video and channel resolvers? So just get the number of views/follows directly in the stitching resolver? Then we could just remove Orion resolvers for that and remove views/follows queries from the Orion schema, just have them as field in the stitched schema. Does that make sense? What do you think?

4. I suggest to rename the query node schema file to something like `queryNodeSchemaExtension` as it contains only additional stuff now. I also noticed it contains all the Orion queries, why is that needed? Those should end up in the final schema anyway because we stitch `orionSchema`, right?

**Pull request closed without merging**